crates/orchestrator/src/tests/jobs/da_job/mod.rs

```diff
@@ -1,64 +1,308 @@
+                internal_id: internal_id.to_string(),
+                job_type: JobType::DataSubmission,
+                status: JobStatus::Created,
+                external_id: ExternalId::String(internal_id.to_string().into_boxed_str()),
+                metadata: HashMap::default(),
+                version: 0,
+            },
+        )
+        .await;
+
+    match response {
+        Ok(_) => {
+            panic!("This testcase's process_job was supposed to throw an error, it successed instead.")
+        }
+        Err(e) => {
+            let expected = eyre!(
+                "Exceeded the maximum number of blobs per transaction: allowed {}, found {} for block {} and job id {}",
+                max_blob_per_txn,
+                current_blob_length,
+                internal_id.to_string(),
+                Uuid::default()
+            )
+            .to_string();
+
+            assert_eq!(e.to_string(), expected);
+        }
+    }
+    state_update_mock.assert();
+    let _ = drop_database().await;
+
+    Ok(())
 }
 
+/// Tests DA Job processing failure when a block is in pending state.
+/// Simulates a pending block state update and expects job processing to fail.
+/// Validates that the error message matches the expected pending state error.
+/// Asserts correct behavior by comparing the received and expected error messages.
 #[rstest]
 #[tokio::test]
-async fn test_verify_job(#[from(default_job_item)] mut job_item: JobItem) {
+async fn test_da_job_process_job_failure_on_pending_block() -> Result<()> {
+    let server = TestConfigBuilder::new().build().await;
+    let config = config().await;
+    let internal_id = "1";
+
+    let pending_state_update = MaybePendingStateUpdate::PendingUpdate(PendingStateUpdate {
+        old_root: FieldElement::default(),
+        state_diff: StateDiff {
+            storage_diffs: vec![],
+            deprecated_declared_classes: vec![],
+            declared_classes: vec![],
+            deployed_contracts: vec![],
+            replaced_classes: vec![],
+            nonces: vec![],
+        },
+    });
+
+    let pending_state_update = serde_json::to_value(&pending_state_update).unwrap();
+    let response = json!({ "id": 1,"jsonrpc":"2.0","result": pending_state_update });
+
+    let state_update_mock = server.mock(|when, then| {
+        when.path("/").body_contains("starknet_getStateUpdate");
+        then.status(200).body(serde_json::to_vec(&response).unwrap());
+    });
+
+    let response = DaJob
+        .process_job(
+            config.as_ref(),
+            &mut JobItem {
+                id: Uuid::default(),
+                internal_id: internal_id.to_string(),
+                job_type: JobType::DataSubmission,
+                status: JobStatus::Created,
+                external_id: ExternalId::String("1".to_string().into_boxed_str()),
+                metadata: HashMap::default(),
+                version: 0,
+            },
+        )
+        .await;
+
+    match response {
```

Comment:
checkout `assert_matches!` for this. it's very useful in test cases for these expressions

Reply:
you can check the old madara repo for examples